src/userSync.js

```diff
@@ -0,0 +1,102 @@
+import * as utils from 'src/utils';
+
+const userSync = exports;
+// A queue of user syncs for each adapter
+const queue = {
+  image: []
+};
+
+/**
+ * @function fireSyncs
+ * @summary Trigger all user syncs in the queue
+ * @private
+ */
+function fireSyncs() {
+  try {
+    // Fire image pixels
+    queue.image.forEach((sync) => {
+      let bidderName = sync[0];
+      let trackingPixelUrl = sync[1];
+      utils.logMessage(`Invoking image pixel user sync for bidder: ${bidderName}`);
+      // insertAdjacentHTML expects HTML string - convert DOM object to string
+      let img = userSync.createImgObject(trackingPixelUrl);
+      if (img) {
+        utils.insertElement(img);
+      }
+    });
+    // Reset the image pixel queue
+    queue.image = [];
+  }
+  catch (e) {
+    utils.logError('Error firing user syncs', e);
+  }
+}
+
+/**
+ * @function hideAndIdElem
+ * @summary Modifies a DOM element to be hidden from user sight and adds a unique ID
+ * @private
+ * @params {object} elementNode A valid DOM element
+ * @returns {object} A valid DOM element
+ */
+function hideAndIdElem(elementNode) {
+  elementNode = elementNode.cloneNode();
+  elementNode.height = 0;
+  elementNode.width = 0;
+  elementNode.style.display = 'none';
+  elementNode.id = utils.getUniqueIdentifierStr();
+  return elementNode;
+}
+
+/**
+ * @function createImgObject
+ * @summary Create an img DOM element for sending a pixel. Made public for test purposes
+ * @public
+ * @params {string} url The URL for the image pixel
+ * @returns {object} A valid DOM element
+ */
+userSync.createImgObject = function(url) {
+  if (!url) {
+    return;
+  }
+  let img = hideAndIdElem(new Image());
```

Comment:
Using the hideAndIdElem function for an in-memory image is really not required, right? I'm guessing that hideAndIdElem was created for iframe support? Which leads me nicely onto my next question: Can iframe support go into this initial implementation? It would make sense if the goal is for bidders to adopt using this util over creating their own routines dealing with user syncing.